### ISSUE TITLE
Version9 Preview A11y Bug: Aria Requiredness is not announced for select button

### ISSUE THREAD

**Issue by hailthekid** [Workday/canvas-kit#2527]
## 🐛 Bug Report
In version 9 for 'canvas-kit-preview-react/select';
Select with the the required prop OR aria-required passed in does not announce requiredness
In Version 10 instead, of a button, select component uses an input that announces aria requiredness when the required is passed in I believe (so no need to pass in aria-required).

## To Reproduce
in version 9, import select component from 'canvas-kit-preview-react/select';
and pass in required prop or aria-required prop
you can see requiredness not announced

## Expected Behavior
requiredness should be announced when passing in required prop

## Actual Results

requiredness is not annouunced 

## Browser (if applicable)

Google chrome

## Link to repl or repo (highly encouraged)

I'll try to post a sandbox io once I figure out how it works
